data-products/src/main/scala/org/sunbird/analytics/exhaust/OnDemandDruidExhaustJob.scala

```diff
@@ -0,0 +1,280 @@
+                                 storageSecret: Option[String] = Option(AppConf.getConfig("storage.secret.config")))
+case class OnDemandDruidResponse(file: List[String], status: String, statusMsg: String, execTime: Long)
+case class FinalMetrics(totalRequests: Option[Int], failedRequests: Option[Int], successRequests: Option[Int])
+
+object OnDemandDruidExhaustJob extends optional.Application with BaseReportsJob with Serializable with OnDemandExhaustJob{
+  override def getClassName() : String = "org.sunbird.analytics.exhaust.OnDemandDruidExhaustJob"
+  val jobName: String = "OnDemandDruidExhaustJob"
+  def name(): String = "OnDemandDruidExhaustJob"
+  // $COVERAGE-OFF$ Disabling scoverage for main and execute method
+  def main(config: String)(implicit sc: Option[SparkContext] = None, fc: Option[FrameworkContext] = None) {
+    JobLogger.init("OnDemandDruidExhaustJob")
+    JobLogger.start("OnDemandDruidExhaustJob Started executing", Option(Map("config" -> config, "model" -> name)))
+    implicit val jobConfig = JSONUtils.deserialize[JobConfig](config)
+    implicit val spark: SparkSession = openSparkSession(jobConfig)
+    implicit val sc: SparkContext = spark.sparkContext
+    implicit val sqlContext = new SQLContext(sc)
+    JobContext.parallelization = CommonUtil.getParallelization(jobConfig)
+
+    implicit val frameworkContext: FrameworkContext = getReportingFrameworkContext()
+    implicit val conf = spark.sparkContext.hadoopConfiguration
+    execute()
+    // $COVERAGE-ON$ Disabling scoverage for main and execute method
+  }
+
+  def validateEncryptionRequest(request: JobRequest): Boolean = {
+    if (request.encryption_key.nonEmpty) true else false;
+  }
+
+  def validateRequest(request: JobRequest): Boolean = {
+    if (request.request_data != null) true else false
+  }
+
+  def getRequests(jobId :String)(implicit spark: SparkSession, fc: FrameworkContext): Array[JobRequest] = {
+    val jobStatus = List("SUBMITTED", "FAILED")
+    val encoder = Encoders.product[JobRequest]
+    val reportConfigsDf = spark.read.jdbc(url, requestsTable, connProperties)
+      .where(col("job_id") === jobId && col("iteration") < 3).filter(col("status").isin(jobStatus: _*));
+    val requests = reportConfigsDf.withColumn("status", lit("PROCESSING")).as[JobRequest](encoder).collect()
+    requests
+  }
+
+  def markRequestAsProcessing(request: JobRequest):Boolean = {
+    request.status = "PROCESSING";
+    updateStatus(request);
+  }
+
+  def getStringProperty(config: Map[String, AnyRef], key: String, defaultValue: String) : String = {
+    config.getOrElse(key, defaultValue).asInstanceOf[String]
+  }
+  def druidAlgorithm(reportConfig: ReportConfig)(implicit spark: SparkSession, sqlContext: SQLContext, fc: FrameworkContext, sc:SparkContext, config: JobConfig): RDD[DruidOutput]  ={
+    val strConfig = JSONUtils.serialize(reportConfig)
+    val reportConfigs = JSONUtils.deserialize[ReportConfig](strConfig)
```

Comment:
Why is reportConfig serialised and again deserialised back to same type **ReportConfig** here? reportConfig can be directly used instead.

Reply:
Fixed this issue